resources/views/posts/show.blade.php

```diff
@@ -5,26 +5,49 @@
         <x-validation-errors :errors="$errors" />
 
         <article class="mb-2">
-            <h2 class="font-bold font-sans break-normal text-gray-900 pt-6 pb-1 text-3xl md:text-4xl break-words">{{ $post->title }}</h2>
+            <h2 class="font-bold font-sans break-normal text-gray-900 pt-6 pb-1 text-3xl md:text-4xl break-words">
+                {{ $post->title }}</h2>
             <h3>{{ $post->user->name }}</h3>
             <p class="text-sm mb-2 md:text-base font-normal text-gray-600">
-                <span class="text-red-400 font-bold">{{ date('Y-m-d H:i:s', strtotime('-1 day')) < $post->created_at ? 'NEW' : '' }}</span>
+                <span
+                    class="text-red-400 font-bold">{{ date('Y-m-d H:i:s', strtotime('-1 day')) < $post->created_at ? 'NEW' : '' }}</span>
                 {{ $post->created_at }}
             </p>
             <img src="{{ $post->image_url }}" alt="" class="mb-4">
             <p class="text-gray-700 text-base">{!! nl2br(e($post->body)) !!}</p>
         </article>
+        
+        @auth
+            @if (auth()->user()->hasLiked($post))
+                <form action="{{ route('posts.likes.destroy', [$post, $post->likes->first()]) }}" method="post">
```

Comment:
PostController.php／showアクションで以下のようにすると、show.blade.phpがスッキリします✨
やっていることは同じです！
ただ、複雑な処理はなるべくコントローラーのほうに記述して、テンプレートでは基本的に変数を受け取るだけにしておいたほうがわかりやすくなります。

```
    public function show($id)
    {
        $post = Post::find($id);

        if (Auth::user()) {
            $like = Like::where('post_id', $post->id)
                ->where('user_id', auth()->user()->id)->first();
            return view('posts.show', compact('post', 'like'));
        } else {
            return view('posts.show', compact('post'));
        }
    }
```

```suggestion
            @if ($like)
                <form action="{{ route('posts.likes.destroy', [$post, $like]) }}" method="post">
```